CONTRIBUTING.md

```diff
@@ -0,0 +1,269 @@
+        1. Description: ***Please be clear and concise***
+        2. Value and Motivation
+            - Feature and/or Functionalities Enabled:
+            - Any Alternatives
+        3. Any additional information:
+
+## Acceptance Criteria ##
+
+The goal of rocWMMA is to provide a C++ API for accelerating matrix multiply accumulate (MMA) operations utilizing AMD GPU hardware.
+Contributors that wish to help optimize and expand the capabilities of rocWMMA in furtherance of this goal should adhere to the following
+guidelines for all features and fixes. Detailed coding style and pull request guidelines are covered later sections.
+
+Contributors wishing to submit new features for rocWMMA should follow the guidelines outlined below:
+
+- Performance Improvements
+    * Features targeting performance improvements for any aspect of rocWMMA are generally permitted. Any optimizations must be both notable
+      and repeatable in order to avoid unnecessary code maintenance. Documentation regarding performance improvements such as benchmark test
+      comparisons must also be provided.
+- Bug Fixes
+    * Any observed unintended behavior within the rocWMMA library should first be documented by filing a bug report issue using the above
+      template. Non-critical issues may be deferred until future releases.
+- WMMA Porting
+    * Developers wishing to implement gap closures with nvcuda::wmma may suggest additional features to do so
+
+All new features and fixes should also tie into the rocWMMA [GitHub Issues](https://github.com/ROCm/rocWMMA/issues) tab:
+
+- **Enhancements**
+    + Any implementations of pre-filed enhancement requests should clearly link to the original issue.
+    + Any new enhancements should be documented as if filing a new enhancement request using the template above.
+- **Bug Fixes**
+    + Any fixes for pre-filed issues should clearly link to the original issue.
+    + Any newly found issues should be documented as if filing a new issue using the template above.
+
+### Exceptions ###
+
+Exceptions to these criteria will be handled on a case-by-case basis, and should be discussed via the Issues tab.
+
+## Code Structure ##
+
+The organization of the rocWMMA library is explained in detail in the [Programmers Guide](https://github.com/ROCm/rocWMMA/blob/develop/docs/Programmers_Guide.rst)
+
+## Coding Style ##
+
+This project follows the [CPP Core Guidelines](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md)
+with few modifications or additions noted below. All pull-requests should in good faith attempt to follow the guidelines stated therein,
+but we recognize that the content is lengthy. Below we list our primary concerns when reviewing pull-requests.
+
+### Interface ###
+
+- Library code should use C++17.
+- Our minimum supported compiler is hipcc 4.4.
+- Avoid CamelCase.
+    * This rule applies specifically to publicly visible APIs, but is also encouraged (not mandated) for internal code
+
```

Comment:
Add bullet point: If you are unsure, inspect surrounding code for consistency and don't be afraid to pose questions for clarification on the PRs.